vortex-serde/src/layouts/read/stream.rs

```diff
@@ -116,21 +120,38 @@ impl<R: VortexReadAt + Unpin + 'static> Stream for LayoutBatchStream<R> {
                     self.state = StreamingState::NextSplit;
                 }
                 StreamingState::NextSplit => {
-                    self.current_selector = self.splits.pop_front().map(|(begin, end)| unsafe {
+                    let new_selector = self.splits.pop_front().map(|(begin, end)| unsafe {
                         RowMask::new_unchecked(
                             Bitmap::from_range(0..(end - begin) as u32),
                             begin,
                             end,
                         )
                     });
 
-                    if self.current_selector.is_none() {
+                    let Some(mut new_selector) = new_selector else {
                         return Poll::Ready(None);
-                    }
+                    };
 
                     self.state = if self.filter_reader.is_some() {
+                        if let Some(indices_mask) = &self.indices_mask {
+                            if indices_mask
+                                .slice(new_selector.begin(), new_selector.end())
+                                .is_empty()
+                            {
+                                self.state = StreamingState::NextSplit;
+                                continue;
+                            }
+                        }
+
+                        self.current_selector = Some(new_selector);
                         StreamingState::Filter
                     } else {
+                        if let Some(indices_mask) = &self.indices_mask {
+                            new_selector.and_inplace(
+                                &indices_mask.slice(new_selector.begin(), new_selector.end()),
+                            )?;
```

Comment:
we still want this to happen in filter case. We should handle non danse selection in filter

Reply:
hmmmm, ok let's file this as an issue. I think this could be a followup

Reply:
filed #1243